### PR TITLE
[Term Entry] Dart Type Conversion .toStringAsPrecision()

### LINKED ISSUE #4567: [Term Entry] Dart Type Conversion .toStringAsPrecision()
### Reason/inspiration (optional)

We would like a new term entry under Dart for the Dart Type Conversion `.toStringAsPrecision()` method.
Provide the entry in a markdown file under `docs/content/dart/concepts/type-conversion/terms/toStringAsPrecision/toStringAsPrecision.md`

The entry should
- Have an introductory paragraph about what the method is and does.
- A syntax section with a pseudo code block with an explanation of it.
- An example showing the method in use.

Please refer to the [entry templates](https://github.com/Codecademy/docs/blob/main/documentation/), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [ ] Issue is assigned

### PR DESCRIPTION
**Pull request by tomasbaublys**
### Description

- Added Dart Type Conversion .toStringAsPrecision()
- Closes Issue ##4567

### Issue Solved
Closes #4567 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

File: content/dart/concepts/type-conversion/terms/toStringAsPrecision/toStringAsPrecision.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+---
+Title: '.toStringAsPrecision()'
+Description: 'Converts this number to a double and returns a string representation of that value.'
```
Comment (cigar-galaxy82) [inline content/dart/concepts/type-conversion/terms/toStringAsPrecision/toStringAsPrecision.md:3]:
```suggestion
Description: 'Converts a number to a double and returns a string representation with the given precision significant digits.'
```

File: content/dart/concepts/type-conversion/terms/toStringAsPrecision/toStringAsPrecision.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+## Example
+
+The following examples illustrates how the `.toStringAsPrecision()` method is utilized to convert this number to a double and returns a string representation of that value:
+
+```dart
+1.toStringAsPrecision(2);
+1e15.toStringAsPrecision(3);
+1234567.toStringAsPrecision(9); 
+```
```
Comment (cigar-galaxy82) [inline content/dart/concepts/type-conversion/terms/toStringAsPrecision/toStringAsPrecision.md:36]: The reason for this in the code block the code should run just by copying and pasting it

File: content/dart/concepts/type-conversion/terms/toStringAsPrecision/toStringAsPrecision.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+## Syntax
+
+```pseudo
+String toStringAsPrecision(int precision);
+```
+The parameter `precision` must be an integer satisfying: 1 <= precision <= 21.
+
+## Example
+
+The following examples illustrates how the `.toStringAsPrecision()` method is utilized to convert this number to a double and returns a string representation of that value:
```
Comment (cigar-galaxy82) [inline content/dart/concepts/type-conversion/terms/toStringAsPrecision/toStringAsPrecision.md:30]:
```suggestion
The following examples illustrate how the `.toStringAsPrecision()` method is utilized to convert this number to a double and returns a string representation of that value:
```

File: content/dart/concepts/type-conversion/terms/toStringAsPrecision/toStringAsPrecision.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  - 'Data Types'
+  - 'Methods'
+  - 'Numbers'
+  - 'Strings'
+CatalogContent:
+  - 'learn-dart'
+  - 'paths/computer-science'
+---
+
+In Dart, **`.toStringAsPrecision()`** is an abstract method used to convert this number to a `double`. This method returns a string representation of that value with exactly precision significant digits.
```
Comment (cigar-galaxy82) [inline content/dart/concepts/type-conversion/terms/toStringAsPrecision/toStringAsPrecision.md:19]:
```suggestion
In Dart, **`.toStringAsPrecision()`** is an abstract method used to convert this number to a `double`. This method returns a string representation of that value with exact precision significant digits.
```

File: content/dart/concepts/type-conversion/terms/toStringAsPrecision/toStringAsPrecision.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+The parameter `precision` must be an integer satisfying: 1 <= precision <= 21.
+
+## Example
+
+The following examples illustrates how the `.toStringAsPrecision()` method is utilized to convert this number to a double and returns a string representation of that value:
+
+```dart
+1.toStringAsPrecision(2);
+1e15.toStringAsPrecision(3);
+1234567.toStringAsPrecision(9);
```
Comment (cigar-galaxy82) [inline content/dart/concepts/type-conversion/terms/toStringAsPrecision/toStringAsPrecision.md:35]:
```suggestion
void main() {
  String a1 = 1.toStringAsPrecision(2);
  print("Output = " + a1);
  String a2 = 1e15.toStringAsPrecision(3);
  print("Output = " + a2);
  String a3 = 1234567.toStringAsPrecision(9);
  print("Output = " + a3);
}
```

**Review by cigar-galaxy82 — CHANGES_REQUESTED** [review]
Hi, I have suggested a few changes

**Comment by cigar-galaxy82** [comment]
Hi @tomasbaublys 

Thank you for contributing to Codecademy docs, the entry is well written I have suggested a few changes please modify

**Comment by mamtawardhani** [comment]
Hey @tomasbaublys! Could you please make the required changes at your earliest convenience?

**Comment by tomasbaublys** [comment]
> Hey @tomasbaublys! Could you please make the required changes at your earliest convenience?

@mamtawardhani I already comited required changes

**Review by cigar-galaxy82 — APPROVED** [review]

**Review by mamtawardhani — APPROVED** [review]
Thank you for the contribution @tomasbaublys!  😃 
The entry is ready to be merged! 🚀

**Pull request merged**